mpas_analysis/sea_ice/time_series.py

```diff
@@ -692,6 +693,10 @@ def _compute_area_vol(self):
             if maxAllowedSeaIceThickness is not None:
                 mask = np.logical_and(mask,
                                       ds.iceThick <= maxAllowedSeaIceThickness)
+            if os.path.exists(outFileNames[hemisphere]):
+                dsCache = xr.open_dataset(outFileNames[hemisphere])
+                timeMask = ds.startTime > dsCache.startTime.isel(Time=-1)
+                mask = np.logical_and(mask, timeMask)
```

Comment:
I don't think this will work as intended if the times actually overlap.  It's fine to mask things in space since we sum over `nCells` but because we don't sum over `Time`, the masked entries will just be `NaN` rather than being removed.  Later, the concat will potentially fail because of overlapping `Time` entries.

I would suggest:

```suggestion
                timeMask = ds.startTime > dsCache.startTime.isel(Time=-1)
                ds = ds.isel(Time=timeMask)
```
I'm pretty confident that `isel` works with a boolean mask as an indexer but it's still worth checking to be sure.

Reply:
Ah, I'll try this instead. Thanks

Reply:
I have retested with a case that has all requested time stamps cached (mask should be all zeros) and the original test with only part of the time period cached and both run successfully.